### ISSUE TITLE
WARNING: QApplication was not created in the main() thread.

### ISSUE THREAD

**Issue by xiaohangguo** [quangtien19999/people_count_by_yolov5#1]
I use the Terminal   ` python app.py --cfg configs/config.yaml`
out 
```
YOLOv5 🚀 v6.1-278-g1ab23fc Python-3.7.0 torch-1.11.0+cu102 CUDA:0 (NVIDIA GeForce RTX 2070, 7960MiB)

Fusing layers... 
YOLOv5n summary: 213 layers, 1867405 parameters, 0 gradients
Adding AutoShape... 
Starting Camera 1
Starting Camera 2
WARNING: QApplication was not created in the main() thread.
QObject::moveToThread: Cannot move objects with a parent
QObject::setParent: Cannot set parent, new parent is in a different thread
QObject::setParent: Cannot set parent, new parent is in a different thread
QMetaMethod::invoke: Dead lock detected in BlockingQueuedConnection: Receiver is GuiReceiver(0x7f95d4000f30)

(process:35755): GLib-CRITICAL **: 13:15:21.219: g_main_context_pop_thread_default: assertion 'g_queue_peek_head (stack) == context' failed
段错误 (核心已转储)

```
don't know what happen
Ubuntu20.04 python3.7 and use your env

**Comment by quangtien19999** [comment]
Hi @xiaohangguo, I think there was a conflict between opencv-python and pyqt. I build this code on Anaconda and Windows 10 and haven't seen this issue, but I found some people with your problem posting on [Stack overflow](https://stackoverflow.com/questions/46449850/how-to-fix-the-error-qobjectmovetothread-in-opencv-in-python). Please try one of these ways like reinstall another version of opencv-python or try using Anaconda. I may help you.

**Comment by xiaohangguo** [comment]
Hi @quangtien19999 , your cat so cute ,jaja…… .  I 'm so happy hear from you through is too late in our country. So,i can't waiting for update my env. As you said, pyqt's version make this error. I update the opencv version to `4.3.0.36` . And I  use  `conda install pyqt`. However,something news go wrong!
I'm freshman with your code , So, give your something out,help me please! Thanks~

Out:
```
YOLOv5 🚀 v6.1-278-g1ab23fc Python-3.7.0 torch-1.11.0+cu102 CUDA:0 (NVIDIA GeForce RTX 2070, 7960MiB)

Fusing layers... 
YOLOv5n summary: 213 layers, 1867405 parameters, 0 gradients, 4.5 GFLOPs
Adding AutoShape... 
Starting Camera 1
Starting Camera 2
Gtk-Message: 00:22:33.685: Failed to load module "canberra-gtk-module"
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/home/hang/anaconda3/envs/yolo/lib/python3.7/threading.py", line 917, in _bootstrap_inner
    self.run()
  File "app.py", line 17, in run
    camPreview(self.previewName, self.camID, self.cfg, self.yolo_model)
  File "app.py", line 58, in camPreview
    cv2.LINE_AA)
SystemError: new style getargs format but argument is not a tuple

```

**Comment by quangtien19999** [comment]
In [config.yaml](https://github.com/quangtien19999/people_count_by_yolov5/blob/4528ae12ca8d66354fab30b4449899f6a920bd0e/configs/config.yaml), some parameters are lists, which use [] instead of tuple (). In older opencv version, lists are not allowed, so you need to convert lists into tuples by `tuple = tuple(list)`. In [app.py](https://github.com/quangtien19999/people_count_by_yolov5/blob/4528ae12ca8d66354fab30b4449899f6a920bd0e/app.py), you can try converting like this:
From: https://github.com/quangtien19999/people_count_by_yolov5/blob/4528ae12ca8d66354fab30b4449899f6a920bd0e/app.py#L50-L58 Into:
`            frame = cv2.putText(
                frame,
                f'Person count: {len(results)}',
                tuple(plot_cfg['ppl_count_text_pos']),
                font,
                plot_cfg['font_scale'],
                tuple(plot_cfg['ppl_count_text_color']),
                plot_cfg['thickness'],
                cv2.LINE_AA)`

Or you can simlpy update opencv to the latest version `4.6.0.66`, it will accept both lists and tuples. I'm using `4.5.4.58` and it still works for me.

Btw, thank you for reporting these issues, I will update my code to fix them.

**Comment by xiaohangguo** [comment]
Oh! No! I update opencv version to `4.6.0.66` and `4.5.4.58`. The same out for me

```
 YOLOv5 🚀 v6.1-278-g1ab23fc Python-3.7.0 torch-1.11.0+cu102 CUDA:0 (NVIDIA GeForce RTX 2070, 7960MiB)

Fusing layers... 
YOLOv5n summary: 213 layers, 1867405 parameters, 0 gradients, 4.5 GFLOPs
Adding AutoShape... 
Starting Camera 1
Starting Camera 2
WARNING: QApplication was not created in the main() thread.
段错误 (核心已转储)

```

**Comment by quangtien19999** [comment]
I think you should remain your old opencv version `4.3.0.36` and try converting list to tuple as I mentioned before.

**Comment by xiaohangguo** [comment]
I try it ,i update the show people count and show fps.
Now no error out ,but something else come out.
when I run it , no more windows come.Nothing! No windows!  help me bro @quangtien19999

**Comment by quangtien19999** [comment]
You can try using [config_cam.yaml](https://github.com/quangtien19999/people_count_by_yolov5/blob/4528ae12ca8d66354fab30b4449899f6a920bd0e/configs/config_cam.yaml) to show your webcam, not my sample videos.

**Comment by xiaohangguo** [comment]
All right,get it, but what is webcam? I don't have a camera configured myself, do you know some public cameras or websites?

**Comment by quangtien19999** [comment]
Sadly not, can you try another videos from Internet?

**Comment by xiaohangguo** [comment]
I can try it, but the problem I encounter now seems to be a problem with pyqt. The version of pyqt may still have some problems. It is a bit incompatible on ubuntu. After the window pops up, there is no normal video display. I will replace the video according to what you said. see if it changes

I see that some projects can directly process the files that generate mp4. Can this be implemented in your project?

**Comment by quangtien19999** [comment]
Yeah, I think the pyqt is the main reason, you should install another version or update to the latest.

**Comment by xiaohangguo** [comment]
All right ,I 'll try it .By the way ,your cat real so  cute.

**Comment by quangtien19999** [comment]
Haha, thank you. Don't forget to report your problems to me 😄

**Comment by xiaohangguo** [comment]
Ok,I'll .Haha, thank you for replying to me every time, you and I should have a jet lag, it is already early morning in our country, is it noon in your side, when I get up tomorrow, try a downgrade version, and then share the program to run situation to you.

**Comment by quangtien19999** [comment]
It's only midnight here, feel free to ask every time. I'll give you answer after finishing my work. By the way, I don't think my simple project can cause you lots of troubles. I will fix all of them when I have time.

**Comment by xiaohangguo** [comment]
Haha, ok, no problem. Is it late at night on your side? I’m sorry to bother you. We stay up all night to write code every night. Today, my computer is training the yolo model. I can't do it tonight. I will train it tomorrow morning. I will try to test the updated version. If there is any news, I will leave a message for you.
